### PR TITLE
hotUpdate cacheKey

### PR DESCRIPTION
**Pull request by g9d**

### CODE REVIEW

File: src/server/data/collection.js [hunk 1]
```diff
@@ -221,6 +222,7 @@ export class QDataCollection {
 
     dropCachedDbInfo() {
         this.indexesRefreshTime = Date.now();
+        this.provider.isHotUpdate = true;
```
Comment (melsomino) [inline src/server/data/collection.js:225]: remove this

File: src/server/data/collection.js [hunk 2]
```diff
@@ -698,6 +700,12 @@ export class QDataCollection {
         if (this.isTests) {
             return;
         }
+        if (this.provider.isHotUpdate) {
```
Comment (melsomino) [inline src/server/data/collection.js:703]: We must not start providers hot update from this point. We must start it from QData.dropCachedDbInfo (data.js):
this.providers.immutable.hotUpdate()

**Review by melsomino — CHANGES_REQUESTED** [review]
see comments

File: src/server/data/data-provider.js [hunk 3]
```diff
@@ -27,6 +28,14 @@ export interface QDataProvider {
 
     getCollectionIndexes(collection: string): Promise<QIndexInfo[]>;
 
+    getCollectionsForSubscribe(): string[];
+
+    loadFingerprint(collections: string[]): Promise<any>;
```
[the remark this answers is not in the document]
Comment (g9d) [reply src/server/data/data-provider.js:33]: How should I get collections from ArangoProvider?

File: src/server/data/collection.js (hunk 2 above)
Comment (g9d) [reply src/server/data/collection.js:703]: Where should I call `hotUpdate()` for the first time at the start to init fingerprint?

File: src/server/data/data-provider.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (melsomino) [reply src/server/data/data-provider.js:33]: You can request all db collections using Arango API.

File: src/server/data/collection.js (hunk 2 above)
Comment (melsomino) [reply src/server/data/collection.js:703]: QData.start

**Review by melsomino — APPROVED** [review]

**Pull request merged**